### PR TITLE
Placeholder setups and Multisig Build 2 deployment

### PR DESCRIPTION
**Pull request by Michael-A-Heuer**
## Description

This PR adds a `PlaceholderSetup` that we can use to populate plugin repos with earlier versions to maintain the same release and build numbers on multiple chains/layers while saving gas.
Here, we use it to not upload build 1 of the `MultisigSetup` again when deploying to polygon.

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

File: packages/contracts/test/framework/plugin/plugin-repo.ts [hunk 1]
```diff
@@ -373,13 +411,6 @@ describe('PluginRepo', function () {
         RELEASE_METADATA
       );
 
-      await pluginRepo.createVersion(
```
Comment (mathewmeconry) [inline packages/contracts/test/framework/plugin/plugin-repo.ts:376]: What was wrong with this test?

**Review by mathewmeconry — COMMENTED** [review]
I would have chosen a different name than "dummy". Something like UndeployedVersion, VersionPlaceholder, or something when used throws a telling error to the user.

File: packages/contracts/deploy/30_plugins/10_plugin-repos/01_create_address_list_voting_repo_conclude.ts [hunk 2]
```diff
@@ -4,10 +4,10 @@ import {
   PluginRepo__factory,
 } from '../../../typechain';
 import {EHRE} from '../../../utils/types';
-import {getContractAddress} from '../../helpers';
+//import {getContractAddress} from '../../helpers';
```
Comment (mathewmeconry) [inline packages/contracts/deploy/30_plugins/10_plugin-repos/01_create_address_list_voting_repo_conclude.ts:7]: remove unused code

File: packages/contracts/deploy/30_plugins/10_plugin-repos/01_create_address_list_voting_repo_conclude.ts [hunk 3]
```diff
@@ -19,21 +19,20 @@ const func: DeployFunction = async function (hre: EHRE) {
     PluginRepoFactoryDeployment.address,
     deployer
   );
-  const managingDaoAddress = await getContractAddress('DAO', hre);
 
-  const pluginRepoBase = await pluginRepoFactory.pluginRepoBase();
+  //const managingDaoAddress = await getContractAddress('DAO', hre);
```
Comment (mathewmeconry) [inline packages/contracts/deploy/30_plugins/10_plugin-repos/01_create_address_list_voting_repo_conclude.ts:23]: remove unused code

File: packages/contracts/src/plugins/governance/multisig/release-metadata.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "Multisig",
+  "name": "",
```
Comment (mathewmeconry) [inline packages/contracts/src/plugins/governance/multisig/release-metadata.json:2]: why do you remove the name?

File: packages/contracts/src/plugins/placeholder-version/PlaceholderSetup.sol [hunk 5]
```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity 0.8.17;
+
+import {PermissionLib} from "../../core/permission/PermissionLib.sol";
+import {PluginSetup, IPluginSetup} from "../../framework/plugin/setup/PluginSetup.sol";
+
+/// @title PlaceholderSetup
+/// @author Aragon Association - 2023
+/// @notice The setup contract acting as a placeholder for outdated plugin builds. These are used to save gas when giving builds and releases identical plugin repos after deployment that do not need to be published when deploying a plugin repo to a new chain or layer.
```
Comment (mathewmeconry) [inline packages/contracts/src/plugins/placeholder-version/PlaceholderSetup.sol:10]: it doesn't really save gas but prevent users from installing an outdated plugin version

**Review by mathewmeconry — CHANGES_REQUESTED** [review]

File: packages/contracts/deploy/30_plugins/10_plugin-repos/01_create_address_list_voting_repo_conclude.ts (hunk 2 above)
Comment (Michael-A-Heuer) [reply packages/contracts/deploy/30_plugins/10_plugin-repos/01_create_address_list_voting_repo_conclude.ts]: Fixed: 43bfd49e52eb7cfc4b191a3efe45bbe5c4b27d3a

File: packages/contracts/deploy/30_plugins/10_plugin-repos/01_create_address_list_voting_repo_conclude.ts (hunk 3 above)
Comment (Michael-A-Heuer) [reply packages/contracts/deploy/30_plugins/10_plugin-repos/01_create_address_list_voting_repo_conclude.ts]: Fixed in 43bfd49e52eb7cfc4b191a3efe45bbe5c4b27d3a

File: packages/contracts/src/plugins/governance/multisig/release-metadata.json (hunk 4 above)
Comment (Michael-A-Heuer) [reply packages/contracts/src/plugins/governance/multisig/release-metadata.json]: Fixed in 5e13df205a68d9231bca8c6cf1b57d4d614d4480

File: packages/contracts/deploy/30_plugins/00_plugin-setups/40_placeholder_setup.ts [hunk 6]
```diff
@@ -17,7 +18,7 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
 
   const {network} = hre;
 
-  hre.placeholderBuildCIDPath = await uploadToIPFS(
+  (hre as EHRE).placeholderBuildCIDPath = await uploadToIPFS(
```
Comment (mathewmeconry) [inline packages/contracts/deploy/30_plugins/00_plugin-setups/40_placeholder_setup.ts:21]: change it on line 8 that the type of `hre` is  `EHRE`. Then you don't need to typecast

**Review by mathewmeconry — CHANGES_REQUESTED** [review]

File: packages/contracts/src/plugins/placeholder-version/PlaceholderSetup.sol (hunk 5 above)
Comment (Michael-A-Heuer) [reply packages/contracts/src/plugins/placeholder-version/PlaceholderSetup.sol]: Fixed in 95d57e1695705286cec906153defb7a40e06f408

File: packages/contracts/deploy/30_plugins/00_plugin-setups/40_placeholder_setup.ts (hunk 6 above)
Comment (Michael-A-Heuer) [reply packages/contracts/deploy/30_plugins/00_plugin-setups/40_placeholder_setup.ts:23]: Fixed c6f4987dadef3e215fd679d8b32ddf3df715ab8c

**Review by mathewmeconry — DISMISSED** [review]

**Review by mathewmeconry — APPROVED** [review]

**Pull request merged**